### PR TITLE
Dx/add compile hooks

### PR DESCRIPTION
**Pull request by pvbergen**

### CODE REVIEW

File: src/Service/CompilationService.php [hunk 1]
```diff
@@ -210,6 +221,9 @@ public function watch($ttl = 60) {
   public function compile($continueOnError = FALSE, $verbose = FALSE) {
     $this->pauseWatch();
     $this->startCompilation();
+
+    \Drupal::moduleHandler()->invokeAll('iq_scss_compiler_pre_compile', [&$this]);
```
Comment (stefanospetrakis) [inline src/Service/CompilationService.php]: Why use the `&` operator?

File: src/Service/CompilationService.php [hunk 2]
```diff
@@ -258,6 +277,8 @@ public function compile($continueOnError = FALSE, $verbose = FALSE) {
     }
     $this->iterator->rewind();
 
+    \Drupal::moduleHandler()->invokeAll('iq_scss_compiler_post_compile', [&$this]);
```
Comment (stefanospetrakis) [inline src/Service/CompilationService.php]: Dito

File: src/Service/CompilationService.php [hunk 3]
```diff
@@ -37,6 +37,14 @@ class CompilationService {
    */
   protected $logger = NULL;
 
+
+  /**
+   * Whether the call originated on cli.
+   *
+   * @var bool
+   */
+  protected $isCli = FALSE;
```
Comment (stefanospetrakis) [inline src/Service/CompilationService.php]: How about this?
```suggestion
  protected $isCli = PHP_SAPI === 'cli';
```

File: iq_scss_compiler.api.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * @param CompilationService $compilationService
+ *   The active compilation service.
+ */
+function hook_iq_scss_compiler_pre_compile(CompilationService $compilationService)
+{
+}
+
+/**
+ * Called before the compilation starts.
```
Comment (stefanospetrakis) [inline iq_scss_compiler.api.php]:
```suggestion
 * Called after the compilation completes.
```

**Review by stefanospetrakis — DISMISSED** [review]
I had a look here too

File: src/Service/CompilationService.php [hunk 5]
```diff
@@ -92,6 +89,10 @@ public function addSource(string $directory) {
       $this->iterator->append($recursiveIterator);
     }
   }
+  
+  public function isCli() {
+    return $this->isCli();
```
Comment (stefanospetrakis) [inline src/Service/CompilationService.php:94]: @pvbergen Recursion alert!

Comment (pvbergen) [reply src/Service/CompilationService.php:94]: Argh.. I'm too tired today :(
Being fixed in https://github.com/iqual-ch/iq_scss_compiler/pull/7

Comment (pvbergen) [reply src/Service/CompilationService.php:94]: And in https://github.com/iqual-ch/iq_scss_compiler/pull/8 for 3.x

**Pull request merged**